gemfirexd/tools/src/test/java/com/pivotal/gemfirexd/jdbc/BugsTest.java

```diff
@@ -8908,4 +8909,36 @@ public void testBug52352() throws Exception {
       TestUtil.shutDown();
     }
   }
+
+  public void testSNAP_2202() {
+    String[] regions = new String[] {
+        "/" + GfxdConstants.IDENTITY_REGION_NAME,
+        "/SCHEMA",
+        "/_SCHEMA",
+        "/__SCHEMA",
+        "/__SCHEMA_",
+        "/__SCH_EMA_",
+        "/_SCH__EMA_",
+        "/__SCH__EMA__",
+        "/SCHEMA/TEST",
+        "/_SCHEMA/TEST",
+        "/__SCHEMA/_TEST",
+        "/__SCHE_MA/TEST",
+        "/__SCHEMA/_TE__ST",
+        // the pattern "_/_" is unsupported
+        // "/__SC__HEMA_/_TE_ST__"
+    };
+    int[] bucketIds = new int[] { 0, 1, 23, 101, 1001 };
+    for (String region : regions) {
+      for (int bucketId : bucketIds) {
+        // below is same as ProxyBucketRegion.fullPath initialization
+        String fullPath = Region.SEPARATOR +
```

Comment:
For escaped path "/__PR/_B______IDENTITYREGION2_10" the unescaped path is coming as "/_PR//B___IDENTITYREGION2/10"

This is right?

Reply:
/__PR/_B_ is removed separately by PartitionedRegionHelper.getPRPath. You see a problem with the test?

Reply:
And yes escaped path can never have slashes in it.

Reply:
Yep.